### PR TITLE
Dropbox - Delete File or Folder

### PR DESCRIPTION
**Pull request by mihaitusa**
Operation with unit tests and refining fallback api version mechanism.
Signed off by Tusa Mihai: tusa.mihai@gmail.com

### CODE REVIEW

File: cs-dropbox/src/main/java/io/cloudslang/content/dropbox/factory/folders/FolderHelper.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ *
+ * The Apache License is available at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *******************************************************************************/
+package io.cloudslang.content.dropbox.factory.folders;
+
+import io.cloudslang.content.dropbox.entities.inputs.InputsWrapper;
+
+class FolderHelper {
```
Comment (Bonczidai) [inline cs-dropbox/src/main/java/io/cloudslang/content/dropbox/factory/folders/FolderHelper.java]: how about creating here a FolderBuilder instead which get params like type(create / delete), autoRename etc..

File: cs-dropbox/src/main/java/io/cloudslang/content/dropbox/actions/folders/CreateFolder.java [hunk 2]
```diff
@@ -145,7 +141,6 @@
             })
     public Map<String, String> execute(@Param(value = ENDPOINT, required = true) String endpoint,
                                        @Param(value = ACCESS_TOKEN, required = true, encrypted = true) String accessToken,
-                                       @Param(value = API_VERSION) String apiVersion,
```
Comment (Bonczidai) [inline cs-dropbox/src/main/java/io/cloudslang/content/dropbox/actions/folders/CreateFolder.java]: hmm - Y we don't need the version anymore?

File: cs-dropbox/src/main/java/io/cloudslang/content/dropbox/entities/dropbox/SuffixUri.java [hunk 3]
```diff
@@ -16,7 +16,10 @@
  * 5/30/2017.
  */
 public enum SuffixUri {
-    CREATE_FOLDER("CreateFolder", "/create_folder");
+    CREATE_FOLDER("CreateFolder", "/create_folder"),
+    CREATE_FOLDER_V2("CreateFolder2", "/create_folder_v2"),
+    DELETE_FILE_OR_FOLDER("DeleteFileOrFolder", "/delete"),
+    DELETE_FILE_OR_FOLDER_V2("DeleteFileOrFolder2", "/delete_v2");
```
Comment (Bonczidai) [inline cs-dropbox/src/main/java/io/cloudslang/content/dropbox/entities/dropbox/SuffixUri.java]: interesting URLs - is this the official URL used in their APIs? is this rest?

File: cs-dropbox/src/main/java/io/cloudslang/content/dropbox/factory/folders/FolderHelper.java (hunk 1 above)
Comment (mihaitusa) [inline cs-dropbox/src/main/java/io/cloudslang/content/dropbox/factory/folders/FolderHelper.java]: @Bonczidai It's a good ideea... sure...

File: cs-dropbox/src/main/java/io/cloudslang/content/dropbox/actions/folders/CreateFolder.java (hunk 2 above)
Comment (mihaitusa) [inline cs-dropbox/src/main/java/io/cloudslang/content/dropbox/actions/folders/CreateFolder.java]: @Bonczidai It has already a fallback mechanism: if user don't provide version then will go on latest/actual one v2, if specified deprecated one then it goes on deprecated api... all that as a result of a previous review....

File: cs-dropbox/src/main/java/io/cloudslang/content/dropbox/entities/dropbox/SuffixUri.java (hunk 3 above)
Comment (mihaitusa) [inline cs-dropbox/src/main/java/io/cloudslang/content/dropbox/entities/dropbox/SuffixUri.java]: yes... is official and it goes like this: deprecated api will consume "/create_folder" uri and the actual api will consume "/create_folder_v2". See here: https://www.dropbox.com/developers/documentation/http/documentation#files-create_folder

**Comment by mihaitusa** [comment]
All... as a remark: the Dropbox API is in the most used 10 API's around the world... so IMHO is worth give it a try...

**Comment by Bonczidai** [comment]
sure - agree

adding @tethryus @andreivasiletruta -  such content worth the attention

**Pull request merged**